subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/TestArtifactsProviderFactory.kt

```diff
@@ -36,19 +29,13 @@ public object TestArtifactsProviderFactory {
 
     // Android API's are unavailable here
     @Suppress("SdCardPath")
-    public fun createForAdbAccess(
-        api: Int,
-        appUnderTestPackage: String,
+    public fun createForAdb(
+        appDirProvider: ApplicationDirProvider,
         name: TestName,
     ): TestArtifactsProvider {
-        val dataPath = if (api >= 30) {
-            "/storage/emulated/0/Android/media/$appUnderTestPackage"
-        } else {
-            "/sdcard/Android/data/$appUnderTestPackage/files"
-        }
```

Comment:
Hide this knowledge in a new abstraction - `ApplicationDirProvider`.
This abstraction is used in a Gradle for adb and in AndroidJunitRunner.